File: removePalindromicSubsequences.py
# problem: https://leetcode.com/problems/remove-palindromic-subsequences/
# Runtime: 32 ms, faster than 58.63% of Python3 online submissions for Remove Palindromic Subsequences.
# Memory Usage: 14.1 MB, less than 90.00% of Python3 online submissions for Remove Palindromic Subsequences.


class Solution:
    def removePalindromeSub(self, s: str) -> int:
        if not s:
            return 0
        if s == s[::-1]:
            return 1
        return 2

File: removePalindromicSubsequences.go
// problem: https://leetcode.com/problems/remove-palindromic-subsequences/
// Runtime: 0 ms, faster than 100.00% of Go online submissions for Remove Palindromic Subsequences.
// Memory Usage: 1.9 MB, less than 84.62% of Go online submissions for Remove Palindromic Subsequences.

package algorithms

func removePalindromeSub(s string) int {
	if len(s) == 0 {
		return 0
	}
	for i, j := 0, len(s)-1; i < j; i++ {
		if s[i] != s[j] {
			return 2
		}
		j--
	}
	return 1
}
